cmake/dependencies.cmake

```diff
@@ -53,6 +53,11 @@ fetch_dep(rapidjson
   TAG 14a5dd756e9bef26f9b53d3b4eb1b73c6a1794d5
   SOURCE_SUBDIR redpanda_build)
 
+fetch_dep(unordered_dense
+  REPO https://github.com/redpanda-data/unordered_dense
+  TAG redpanda
```

Comment:
maybe better to use a SHA or immutable tag? 

otherwise if you pull an old version of this file it will still use the current `redpanda` tag which is probably wrong

Reply:
I went with this as that is what we are doing for some of our other forks (e.g.: seastar, croaring) but we are not consistent. I am happy either way.

Reply:
We should definitely use the sha1 in 3rdparty.cmake @ vtools. Here in the open-source build dependencies are more loose. But they probably shouldn't be. I'll make it a point to go add sha1's here as well.

Reply:
In vtools it's the sha. I will change the OSS one to use the sha as well then if we want to change the other ones as well.